client/src/templates/Challenges/quiz/show.tsx

```diff
@@ -111,147 +165,148 @@ class ShowQuiz extends Component<ShowQuizProps, ShowQuizState> {
       helpCategory
     });
     challengeMounted(challengeMeta.id);
-    this.container.current?.focus();
-  }
+    container.current?.focus();
+    // This effect should be run once on mount
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
-  componentDidUpdate(prevProps: ShowQuizProps): void {
-    const {
-      data: {
-        challengeNode: {
-          challenge: { title: prevTitle }
-        }
-      }
-    } = prevProps;
-    const {
-      challengeMounted,
-      data: {
-        challengeNode: {
-          challenge: { title: currentTitle, challengeType, helpCategory }
+  useEffect(() => {
+    updateChallengeMeta({
+      ...challengeMeta,
+      title,
+      challengeType,
+      helpCategory
+    });
+    challengeMounted(challengeMeta.id);
+  }, [
+    title,
+    challengeMeta,
+    challengeType,
+    helpCategory,
+    challengeMounted,
+    updateChallengeMeta
+  ]);
+
+  const handleAnswersCheck = () => {
+    const unansweredQuestions = quizData.reduce(
+      (accumulator, currentQuestion, currentIndex) => {
+        if (currentQuestion.selectedAnswer == null) {
+          accumulator.push(currentIndex + 1);
         }
+        return accumulator;
```

Comment:
```suggestion
    const unansweredQuestions = quizData.reduce<number[]>(
      (acc, curr, id) => (curr.selectedAnswer == null ? [...acc, id + 1] : acc),
```
Style nit. Slightly cleaner, imo :man_shrugging: